### ISSUE TITLE
Upgrading a package that removes dependencies between solutions is not possible

### ISSUE THREAD

**Issue by ewingjm** [Capgemini/powerapps-packagedeployer-template#30]
## Describe the bug

We have two solutions in the below package. Solution B is dependent on Solution A as it includes a field created in solution A in one of its forms:

**Solution A**
```
|__ Table A (unmanged)
    |__ Field A (unmanaged)
```

**Solution B**
```
|__ Table A (managed)
    |__ Form A (unmanaged)
        |_ Field A (managed)
```

This will deploy fine. In a subsequent commit, we remove Field A from Table A and Form A (thus removing the dependency from Solution B on Field A). Deploying this package version fails due to the dependency on Field A still existing in Solution B. The deployment attempts to remove Field A before it installs the upgraded version of Solution B which removes the dependency.

## To reproduce

- Import [Package with dependency.zip](https://github.com/Capgemini/powerapps-packagedeployer-template/files/5868098/Package.with.dependency.zip)
- Import [Package without dependency.zip](https://github.com/Capgemini/powerapps-packagedeployer-template/files/5868119/Package.without.dependency.zip)

## Expected behaviour

The package should be able to import without dependencies issues (e.g. by importing all upgrades before applying the upgrades in reverse order).

## Additional context

This appears to be a problem with the Package Deployer rather than the template itself but I believe it would be possible implement a workaround until the Package Deployer is fixed. 

## Workaround

The current workaround would be to ensure that you produce separate builds for removing dependencies and then deleting them.

**Comment by ewingjm** [comment]
I believe I raised this with the owners of [https://www.powershellgallery.com/packages/Microsoft.Xrm.Tooling.PackageDeployment.Powershell](Microsoft.Xrm.Tooling.PackageDeployment.Powershell) on the PowerShell Gallery but I didn't get a confirmation despite requesting one. Currently seeing this banner so may not have gone through:

![image](https://user-images.githubusercontent.com/34515225/105738700-e437d080-5f69-11eb-9f58-2f0e2d53c5ab.png)

Will follow up with a link to this issue once the banner disappears.

**Comment by tdashworth** [comment]
@ewingjm This issue has been stale for a while - have you got any updates on this?

**Comment by filcole** [comment]
Hi @tdashworth.  I believe this is expected behaviour of the package deployer, when packages are installed in 'Upgrade' mode.   The work around that is potentially possible is to deprecate the component from solution A, and then remove it from solution B.  A later deployment is to finally remove the field from solution A.

Alternatively, solution A could be deployed in Update mode (not upgrade) so that the field is not deleted from the environment.

**Comment by tdashworth** [comment]
Thanks for that insight @filcole!

**Comment by ewingjm** [comment]
Hi @tdashworth @filcole. Yeah, probably worth just closing the issue. It's a Package Deployer problem rather than related to anything that this project can solve. 

It's a bit tedious being unable to remove both the dependencies and the components in a single release when the solution for this would just be for the Package Deployer to stage all solutions first and apply them in reverse order. That said, I think we've got a more mature understanding of releases now and we rarely do upgrades anyway. It's quite costly as it increases deployment times by about 100% - not great if you're deploying to production regularly due to the additional downtime. The end result is we'll likely never get this issue again as we hardly do upgrades and, when we do, they're done well after all the dependencies of a component have been removed.
